frontend/src/app/state/application/selectors/exercise.selectors.ts

```diff
@@ -212,34 +212,50 @@ export function createSelectByPredicate<E extends WithPosition>(
     );
 }
 
+export function createSelectBehaviorStates(simulatedRegionId: UUID) {
+    return createSelector(
+        createSelectSimulatedRegion(simulatedRegionId),
+        (simulatedRegion) =>
+            simulatedRegion.behaviors as ExerciseSimulationBehaviorState[]
+    );
+}
+
+export function createSelectActivityStates(simulatedRegionId: UUID) {
+    return createSelector(
+        createSelectSimulatedRegion(simulatedRegionId),
+        (simulatedRegion) =>
+            simulatedRegion.activities as {
+                [key: UUID]: ExerciseSimulationActivityState;
+            }
+    );
+}
+
 export function createSelectBehaviorState<
     B extends ExerciseSimulationBehaviorState
 >(simulatedRegionId: UUID, behaviorId: UUID) {
     return createSelector(
-        createSelectSimulatedRegion(simulatedRegionId),
-        (simulatedRegion) =>
-            simulatedRegion?.behaviors.find(
-                (behavior) => behavior.id === behaviorId
-            ) as B | undefined
+        createSelectBehaviorStates(simulatedRegionId),
+        (behaviors) =>
+            behaviors.find((behavior) => behavior.id === behaviorId) as B
     );
 }
 
 export function createSelectActivityState<
     B extends ExerciseSimulationActivityState
```

Comment:
(Although you did not change this line.) Could you please use an `A` (like `activity`) here? The `B` seems like something related to behaviors.